lamindb/models/schema.py

```diff
@@ -579,11 +588,27 @@ def coerce_dtype(self) -> bool:
 
     @coerce_dtype.setter
     def coerce_dtype(self, value: bool) -> None:
-        if self._aux is None:  # type: ignore
-            self._aux = {}  # type: ignore
-        if "af" not in self._aux:
-            self._aux["af"] = {}
-        self._aux["af"]["0"] = value
+        self._aux = self._aux or {}
+        self._aux.setdefault("af", {})["0"] = value
+
+    @property
+    def optional(self) -> QuerySet:
+        """Optional features."""
+        if self._aux is not None and "af" in self._aux and "1" in self._aux["af"]:  # type: ignore
+            return (
+                self.members.filter(uid__in=self._aux["af"]["1"])
+                .order_by("links_schema__id")
+                .all()
+            )
+        else:
+            return Feature.objects.none()  # empty QuerySet
+
+    @optional.setter
+    def optional(self, value: list[Record]) -> None:
```

Comment:
I didn't find the test for this.

Like with all other schema properties, we should list out all datasets that would violate the change.

Say, if somebody makes a feature required that was previously optional, certain datasets that were previously validated would no longer validate.

This is all super complicated and I wonder whether we can take a pragmatic approach by printing a warning instead of trying to query all datasets and testing whether features were linked or not.